src/turndown.js

```diff
@@ -6,6 +6,24 @@ import Node from './node'
 var reduce = Array.prototype.reduce
 var leadingNewLinesRegExp = /^\n*/
 var trailingNewLinesRegExp = /\n*$/
+const markdownReplacements = [
+  [/\\/g, '\\\\'],
+  [/\*/g, '\\*'],
+  [/-/g, '\\-'],
+  [/\+/g, '\\+'],
+  [/=/g, '\\='],
+  [/#/g, '\\#'],
+  [/`/g, '\\`'],
+  [/~/g, '\\~'],
+  [/\|/g, '\\|'],
+  [/\(/g, '\\('],
+  [/\)/g, '\\)'],
+  [/\[/g, '\\['],
+  [/\]/g, '\\]'],
+  [/>/g, '\\>'],
+  [/_/g, '&lowbar;'],
```

Comment:
Yeah this is not great on our part. As part of our stack we use `turndown` to turn HTML to markdown and then later [`marked`](https://github.com/markedjs/marked) to turn the markdown back to HTML.

We were having a weird issue where escaping `_` as `\_` was not then being parsed correctly when going back to HTML. Probably a double escaping issue as the `\` was being rendered to the screen (weirdly we don't have this issue any other time we escape using `\`). Our solution for now in our code is to use `&lowbar;` instead.

Obviously we shouldn't let our specific problem affect `turndown` but we left it there for now whilst we were copying our code here.

So there's probably three options:
* leave it how it is (using `&lowbar;`), at least for now.
* Use HTML character codes everywhere instead of `\` for escaping. A little outlandish but maybe makes sense since the output is HTML anyway.
* Use `\_` in `turndown` instead and we find another approach to our issue. This probably involves us sticking with overloading `escape` for now despite this PR so from that point of view it seems a shame.

Sorry for the long reply. Any thoughts either way?